src/distance_transform.rs

```diff
@@ -95,8 +113,53 @@ pub(crate) enum DistanceFrom {
 }
 
 pub(crate) fn distance_transform_impl(image: &mut GrayImage, norm: Norm, from: DistanceFrom) {
+    match norm {
+        Norm::LInf => distance_transform_impl_linf(image, from),
+        Norm::L1 => distance_transform_impl_l1(image, from),
+        Norm::L2 => {
+            match from {
+                DistanceFrom::Foreground => (),
+                DistanceFrom::Background => image
+                    .iter_mut()
+                    .for_each(|p| *p = if *p == 0 { 1 } else { 0 }),
+            }
+            let float_dist: ImageBuffer<Luma<f64>, Vec<f64>> =
+                euclidean_squared_distance_transform(image);
+            image
+                .iter_mut()
+                .zip(float_dist.iter())
+                .for_each(|(u, v)| *u = v.sqrt().clamp(0.0, 255.0).ceil() as u8);
+        }
+    }
+}
+
+#[inline(always)]
+fn distance_transform_impl_linf(image: &mut GrayImage, from: DistanceFrom) {
```

Comment:
As each of these functions is used only used I think it’s better to just inline the calls to distance_transform_linf_or_l1 into distance_transform_impl and delete the two wrapper functions.

And there’s no need for the comment explaining why we’re using const bool.